### PR TITLE
Code simplifications

### PR DESCRIPTION
**Pull request by cquiroz**
Some minor simplifications:

* Use `extractAs[A]` instead of `extract().as[A]`
* Remove some parameters on the table columns that default correctly upstream

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala [hunk 1]
```diff
@@ -117,11 +117,11 @@ object Tcs {
 
   @SuppressWarnings(Array("org.wartremover.warts.Overloading"))
   def build[T, P: ClassTag](f: P => Endo[T], k: ItemKey, config: Config): Endo[T] =
-    config.extract(k).as[P].map(f).foldK
+    config.extractAs[P](k).map(f).foldK
 
   @SuppressWarnings(Array("org.wartremover.warts.Overloading"))
   def build[T, P: ClassTag, Q: ClassTag](f: (P, Q) => Endo[T], k1: ItemKey, k2: ItemKey, config: Config): Endo[T] =
-    config.extract(k1).as[P].product(config.extract(k2).as[Q]).map(f.tupled).foldK
+    config.extractAs[P](k1).product(config.extractAs[Q](k2)).map(f.tupled).foldK
```
Comment (tpolecat) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala]: This should be the same as `.mapN(f)`

```scala
(config.extractAs[P](k1), config.extractAs[Q](k2)).mapN(f).foldK
```

File: modules/seqexec/server/src/main/scala/seqexec/server/gpi/GPI.scala [hunk 2]
```diff
@@ -66,100 +66,81 @@ object GPI {
...
   private def gpiAoFlags(config: Config): Either[ExtractFailure, AOFlags] =
     for {
-      useAo      <- config.extract(INSTRUMENT_KEY / USE_AO_PROP).as[java.lang.Boolean]
-      useCal     <- config.extract(INSTRUMENT_KEY / USE_CAL_PROP).as[java.lang.Boolean]
+      useAo      <- config.extractAs[JBoolean](INSTRUMENT_KEY / USE_AO_PROP)
+      useCal     <- config.extractAs[JBoolean](INSTRUMENT_KEY / USE_CAL_PROP)
       aoOptimize <- config
-                      .extract(INSTRUMENT_KEY / AO_OPTIMIZE_PROP)
-                      .as[java.lang.Boolean]
+                      .extractAs[JBoolean](INSTRUMENT_KEY / AO_OPTIMIZE_PROP)
```
Comment (tpolecat) [inline modules/seqexec/server/src/main/scala/seqexec/server/gpi/GPI.scala]: Not a big deal but you could remove the leftover newlines in this method and in a few other places.

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala (hunk 1 above)
Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala]: nice

File: modules/seqexec/server/src/main/scala/seqexec/server/gpi/GPI.scala (hunk 2 above)
Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/gpi/GPI.scala]: ok, I've been running scalafmt to ensure the width is under 80, I'll check that again

**Pull request merged**